### PR TITLE
Railgun and tac binos changes

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Railgun takes about 4-5 seconds to reach ground after you start lasing. More lasers dot colors to differenciate tac binos laser mode. Tac binos are also much more convenient to use 

## Why It's Good For The Game

Gives the xenos the necessary time and information to correctly assess the situation. Now, if you get killed by a railgun, it's your own fault. Using tac binos is also much more convenient

## Changelog
:cl:
tweak : tweaked railgunning timing a bit, it now takes time to triangulate the laser and for railgun shells to reach ground (depends on the orbit, between 3-5 seconds to reach ground after being called instead of 1 currently)
tweak: tweaked tac binos, you can know shift click to get coordinates instead of using range finder mode, each mode now has its own laser dot
imageadd: added blue laser dot
fix : firing the OB from tac binos didn't have any randomness like using overwatch did, now fixed
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
railguns are already very weak , another nerf ontop is not needed.

**Comment by BraveMole** [comment]
> railguns are already very weak , another nerf ontop is not needed.

I agree, i'll propose another change to them (much less scatter, much more power) in another PR when i'll understand exactly the code

File: code/game/objects/items/binoculars.dm [hunk 1]
```diff
@@ -16,13 +16,13 @@
...
-#define MODE_RAILGUN 2
-#define MODE_ORBITAL 3
+#define MODE_RAILGUN 1
+#define MODE_ORBITAL 2
+#define MODE_RANGE_FINDER 3
 
 /obj/item/binoculars/tactical
 	name = "tactical binoculars"
-	desc = "A pair of binoculars, with a laser targeting function. Alt+Click to toggle mode. Ctrl+Click when using to target something."
+	desc = "A pair of binoculars, with a laser targeting function. Alt+Click or space to toggle mode. Ctrl+Click when using to target something. Shift+Click to get coordinates. Ctrl+Shift+Click to fire OB when lasing in OB mode"
```
Comment (Wayland-Smithy) [inline code/game/objects/items/binoculars.dm:25]: "space" maybe should be "special action" if it is indeed referring to a bindable hotkey.

Comment (BraveMole) [reply code/game/objects/items/binoculars.dm:25]: Ah yes, mistake

File: code/game/objects/effects/overlays.dm [hunk 2]
```diff
@@ -129,6 +129,96 @@
...
+		squad.squad_laser_targets -= src
+		squad = null
+	if(source_binoc)
+		source_binoc.laser_cooldown = world.time + source_binoc.cooldown_duration
+		source_binoc.laser = null
+		source_binoc = null
+	if(linked_cam)
+		qdel(linked_cam)
+		linked_cam = null
+	. = ..()
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:167]: return ..()

File: code/game/objects/effects/overlays.dm [hunk 3]
```diff
@@ -129,6 +129,96 @@
...
+
+/obj/effect/overlay/temp/laser_target_cas/ex_act(severity) //immune to explosions
+	return
+
+/obj/effect/overlay/temp/laser_target_cas/examine()
+	..()
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:173]: . = ..()

File: code/game/objects/effects/overlays.dm [hunk 4]
```diff
@@ -129,6 +129,96 @@
...
+		squad.squad_laser_targets -= src
+		squad = null
+	if(source_binoc)
+		source_binoc.laser_cooldown = world.time + source_binoc.cooldown_duration
+		source_binoc.laser = null
+		source_binoc = null
+	if(linked_cam)
+		qdel(linked_cam)
+		linked_cam = null
+	. = ..()
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:212]: return ..()

File: code/game/objects/effects/overlays.dm [hunk 5]
```diff
@@ -129,6 +129,96 @@
...
+
+/obj/effect/overlay/temp/laser_target_railgun/ex_act(severity) //immune to explosions
+	return
+
+/obj/effect/overlay/temp/laser_target_cas/examine()
+	..()
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:218]: . = ..()

File: code/game/objects/effects/overlays.dm [hunk 6]
```diff
@@ -129,6 +129,96 @@
...
+	if(ishuman(usr))
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:219]: user not usr

File: code/game/objects/effects/overlays.dm [hunk 7]
```diff
@@ -129,6 +129,96 @@
...
+		linked_cam = null
+	. = ..()
+
+/obj/effect/overlay/temp/laser_target_cas/ex_act(severity) //immune to explosions
+	return
+
+/obj/effect/overlay/temp/laser_target_cas/examine()
+	..()
+	if(ishuman(usr))
+		to_chat(usr, "<span class='danger'>It's a laser to designate cas targets, get away from it!</span>")
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:175]: user not usr

File: code/game/objects/effects/overlays.dm [hunk 8]
```diff
@@ -129,6 +129,96 @@
...
+	name = "laser"
+	anchored = TRUE
+	mouse_opacity = 1
+	icon = 'icons/obj/items/projectiles.dmi'
+	icon_state = "laser_target_coordinate"
+	effect_duration = 600
+	var/target_id
+	var/obj/item/binoculars/tactical/source_binoc
+	var/obj/machinery/camera/laser_cam/linked_cam
+	var/datum/squad/squad
```
Comment (TiviPlus) [inline code/game/objects/effects/overlays.dm:142]: Why are these not children of  laser_target

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**